otter/test/convergence/test_service.py

```diff
@@ -152,34 +505,63 @@ def test_success(self):
             'POST',
             'loadbalancers/23/nodes',
-            data=mock.ANY,
+            data=transform_eq(
+                freeze,
+                pmap({
+                    'nodes': transform_eq(
+                        lambda nodes: set(freeze(nodes)),
+                        set([pmap({'weight': 1, 'type': 'PRIMARY',
+                                   'port': 80,
+                                   'condition': 'ENABLED',
+                                   'address': '10.0.0.2'}),
+                             pmap({'weight': 1, 'type': 'PRIMARY',
+                                   'port': 80,
+                                   'condition': 'ENABLED',
+                                   'address': '10.0.0.1'})]))})),
             success_pred=mock.ANY)
-        got_req = eff.intent.effects[0].intent
-        self.assertEqual(got_req, expected_req.intent)
-        # separate check for nodes; they are unique, but can be in any order
-        self.assertEqual(
-            set(freeze(got_req.data['nodes'])),
-            set([pmap({'weight': 1, 'type': 'PRIMARY', 'port': 80,
-                       'condition': 'ENABLED', 'address': '10.0.0.2'}),
-                 pmap({'weight': 1, 'type': 'PRIMARY', 'port': 80,
-                       'condition': 'ENABLED', 'address': '10.0.0.1'})]))
-        r = resolve_effect(eff, ['stuff'])
-        # The result of the parallel is returned directly
-        # TODO: This must change with issue #844.
-        self.assertEqual(r, ['stuff'])
+        expected_intents = self.expected_intents + [
+            (expected_req.intent, 'successful response')]
+        result = sync_perform(self._get_dispatcher(expected_intents), eff)
+        self.assertEqual(self.group.modify_state_values[-1].active, {})
+        self.assertEqual(result, [(StepResult.SUCCESS, [])])
+
+    def test_first_error_extraction(self):
+        """
+        If the GetScalingGroupInfo effect fails, its exception is raised
+        directly, without the FirstError wrapper.
+        """
+        log = mock_log()
+        gacd = self._get_gacd_func(self.group.uuid)
+        for srv in self.servers:
+            srv.desired_lbs = pmap()
+
+        tscope_eff = execute_convergence(self.tenant_id, self.group_id, log,
+                                         get_all_convergence_data=gacd)
+
+        # Perform the GetScalingGroupInfo by raising an exception
+        dispatcher = ComposedDispatcher([
+            EQFDispatcher([
+                (GetScalingGroupInfo(
+                    tenant_id='tenant-id',
+                    group_id='group-id'),
+                 lambda i: raise_(RuntimeError('foo')))]),
+            self._get_dispatcher()])
+
+        # And make sure that exception isn't wrapped in FirstError.
+        e = self.assertRaises(
+            RuntimeError,
+            sync_perform, dispatcher, tscope_eff.intent.effect)
+        self.assertEqual(str(e), 'foo')
+
+
+def raise_(e):
```

Comment:
+1. Pretty useful. This should be in standard library. Please add it in utils.py. (Non-blocking)